argilla/src/argilla/datasets/_export/_hub.py

```diff
@@ -0,0 +1,194 @@
+            size_categories_parser,
+        )
+
+        hf_api = HfApi()
+
+        hfds = self.records(with_vectors=True, with_responses=True, with_suggestions=True).to_datasets()
+        hfds.push_to_hub(repo_id, *args, **kwargs)
+
+        with TemporaryDirectory() as tmpdirname:
+            self.to_disk(path=tmpdirname, with_records=False)
+            hf_api.upload_folder(
+                folder_path=tmpdirname,
+                repo_id=repo_id,
+                repo_type="dataset",
+                token=kwargs.get("token"),
+                path_in_repo=self._default_dataset_repo_dir_name,
+            )
+
+        if generate_card:
+            sample_argilla_record = next(iter(self.records(with_suggestions=True, with_responses=True)))
+            sample_huggingface_record = hfds[0]
+            card = ArgillaDatasetCard.from_template(
+                card_data=DatasetCardData(
+                    size_categories=size_categories_parser(len(hfds)),
+                    tags=["rlfh", "argilla", "human-feedback"],
+                ),
+                repo_id=repo_id,
+                argilla_fields=self.settings.fields,
+                argilla_questions=self.settings.questions,
+                argilla_guidelines=self.settings.guidelines or None,
+                argilla_vectors_settings=self.settings.vectors or None,
+                argilla_metadata_properties=self.settings.metadata,
+                argilla_record=sample_argilla_record.to_dict(),
+                huggingface_record=sample_huggingface_record,
+            )
+            card.push_to_hub(repo_id, repo_type="dataset", token=kwargs.get("token"))
+
+    @classmethod
+    def from_hub(
+        cls: Type["Dataset"],
+        repo_id: str,
+        workspace: Optional[Union["Workspace", str, UUID]] = None,
+        client: Optional["Argilla"] = None,
+        with_records: bool = True,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        """Loads a `FeedbackDataset` from the Hugging Face Hub.
+
+        Parameters:
+            repo_id: the ID of the Hugging Face Hub repo to load the `Dataset` from.
+            workspace: the workspace to load the `Dataset` into. If not provided, the default workspace will be used.
+            client: the client to use to load the `Dataset`. If not provided, the default client will be used.
+            with_records: whether to load the records from the Hugging Face dataset. Defaults to `True`.
+            *args: the args to pass to `datasets.Dataset.load_from_hub`.
+            **kwargs: the kwargs to pass to `datasets.Dataset.load_from_hub`.
+
+        Returns:
+            A `Dataset` loaded from the Hugging Face Hub.
+        """
+        from datasets import DatasetDict, load_dataset, Dataset
+        from huggingface_hub import hf_hub_download
+
+        with TemporaryDirectory() as tmpdirname:
+            for filename in cls._default_configuration_files:
+                hf_hub_download(
```

Comment:
Better not to do this in a temporary directory. Using `local_dir` will download it do a temporary location. If you don't use this, the cache directory will be used, meaning that next time the files will already be downloaded. So you could do:

```py
from huggingface_hub import snapshot_download

folder_path = snapshot_download( # download both files in parallel
    repo_id=repo_id,
    repo_type="dataset",
    allow_patterns=cls._default_configuration_files
)

dataset = cls.from_disk(path=folder_path, target_workspace=workspace, client=client)
```

Reply:
Maybe use `kwargs.get("token")` btw? (to be consistent with `to_hub`)